vector/v.info/print.c

```diff
@@ -197,31 +234,73 @@ void print_shell(struct Map_info *Map, const char *field_opt)
+        fprintf(stdout, "project=%s\n", G_location());
+        fprintf(stdout, "database=%s\n", G_gisdbase());
+        fprintf(stdout, "title=%s\n", Vect_get_map_name(Map));
+        fprintf(stdout, "scale=1:%d\n", Vect_get_scale(Map));
+        fprintf(stdout, "creator=%s\n", Vect_get_person(Map));
+        fprintf(stdout, "organization=%s\n", Vect_get_organization(Map));
+        fprintf(stdout, "source_date=%s\n", Vect_get_map_date(Map));
+        break;
+    case JSON:
+        json_object_set_string(root_object, "name", Vect_get_name(Map));
+        json_object_set_string(root_object, "mapset", Vect_get_mapset(Map));
+        json_object_set_string(root_object, "location", G_location());
+        json_object_set_string(root_object, "project", G_location());
+        json_object_set_string(root_object, "database", G_gisdbase());
+        json_object_set_string(root_object, "title", Vect_get_map_name(Map));
+        json_object_set_number(root_object, "scale", Vect_get_scale(Map)); // fixme: 1:scale
+        json_object_set_string(root_object, "creator", Vect_get_person(Map));
+        json_object_set_string(root_object, "organization", Vect_get_organization(Map));
+        json_object_set_string(root_object, "source_date", Vect_get_map_date(Map));
+        break;
+    }
+
     /* This shows the TimeStamp (if present) */
     if (time_ok == TRUE && (first_time_ok || second_time_ok)) {
         G_format_timestamp(&ts, timebuff);
-        fprintf(stdout, "timestamp=%s\n", timebuff);
+        switch (format) {
+        case PLAIN:
+            fprintf(stdout, "timestamp=%s\n", timebuff);
+            break;
+        case JSON:
+            json_object_set_string(root_object, "timestamp", timebuff);
+            break;
+        }
     }
     else {
-        fprintf(stdout, "timestamp=none\n");
+        switch (format) {
+        case PLAIN:
+            fprintf(stdout, "timestamp=none\n");
+            break;
+        case JSON:
+            json_object_set_null(root_object, "timestamp");
+            break;
+        }
     }
 
     if (map_type == GV_FORMAT_OGR || map_type == GV_FORMAT_OGR_DIRECT) {
-        fprintf(stdout, "format=%s,%s\n", Vect_maptype_info(Map),
-                Vect_get_finfo_format_info(Map));
-        fprintf(stdout, "ogr_layer=%s\n", Vect_get_finfo_layer_name(Map));
-        fprintf(stdout, "ogr_dsn=%s\n", Vect_get_finfo_dsn_name(Map));
-        fprintf(stdout, "feature_type=%s\n", Vect_get_finfo_geometry_type(Map));
+        switch (format) {
+        case PLAIN:
+            fprintf(stdout, "format=%s,%s\n", Vect_maptype_info(Map),
+                    Vect_get_finfo_format_info(Map));
+            fprintf(stdout, "ogr_layer=%s\n", Vect_get_finfo_layer_name(Map));
+            fprintf(stdout, "ogr_dsn=%s\n", Vect_get_finfo_dsn_name(Map));
+            fprintf(stdout, "feature_type=%s\n", Vect_get_finfo_geometry_type(Map));
+            break;
+        case JSON:
+            // fixme: add format=%,%s
```

Comment:
I am not sure what the maximum lengths of the `Vect_maptype_info` and `Vect_get_finfo_format_info` can be? Is there a utility function in the codebase to concat variable length strings?

Reply:
That's a difficult question, fortunately, the right solution is somewhere else. The format key is two different values one would have to parse out, so it should be two different key-value pairs in JSON.

How to name these two is another question. The schema is sort of like this:

```
format: native|OGR|PostGIS|...
format-detail: NULL|Shapefile?|GeoPackage|...|PostgreSQL
```

The "detail" is applicable only for non-native format. NULL pointer is printed otherwise for plain format (something like `(null)` I guess).

Reply:
Thanks for the explanation, fixed.